docs/source/developers/reviewing.rst

```diff
@@ -0,0 +1,256 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+=======================
+Reviewing contributions
+=======================
+
+Principles
+==========
+
+Arrow is a foundational project that will need to evolve over many years
+or even decades, while serving potentially millions of users.  We believe
+that being meticulous when reviewing brings greater rewards to the project
+than being lenient and aiming for quick merges.
+
+Code review like this leads to better quality code, more people who are
+engaged with and understand the code being changed, and a generally
+healthier project with more room to grow and accomodate emerging needs.
+
+
+Guidelines
+==========
+
+Meta
+----
+
+**Use your own judgement**.  These guidelines are not hard rules.
+Committers are expected to have sufficient expertise on their work
+areas to be able to adjust their approach based on any concerns they have.
+
+These guidelines are not listed in a particular order or not intended to be
+used as a checklist.
+
+Finally, these guidelines are not currently exhaustive.
```

Comment:
```suggestion
Finally, these guidelines are not exhaustive.
```
nit: can you ever have exhaustive guidelines on code reviews given, as mentioned above, the need for discretion?

Reply:
Of course :-D